OPHD/Technology/TechnologyCatalog.h

```diff
@@ -16,11 +16,14 @@ class TechnologyCatalog
 	TechnologyCatalog() = delete;
 	TechnologyCatalog(const std::string& techFile);
 
+	const std::vector<std::string>& categories() const { return mCategorNames; }
+
 	const Technology& technologyFromId(int id) const;
 
 	const std::vector<Technology> technologiesInCategory(const std::string& categoryName) const;
 
 private:
+	std::vector<std::string> mCategorNames;
 	std::map<std::string, std::vector<Technology>> mCategories;

```

Comment:
I opted to have this cached for the UI functions that are going to come with the Research UI. Might as well have those cached so that we can avoid doing any sort of lookups besides basic linear vector traversal.